searchlib/src/main/java/com/yahoo/searchlib/rankingexpression/ExpressionFunction.java

```diff
@@ -151,6 +153,15 @@ public Instance expand(SerializationContext context, List<ExpressionNode> argume
         return new Instance(symbol, expressionString);
     }
 
+    private boolean shouldGenerateFeature(ExpressionNode expr) {
+        if (expr instanceof ConstantNode) return false;
+        if (expr instanceof ReferenceNode) return false;
+        if (expr instanceof NameNode) return false;
+        if (expr instanceof FunctionNode) return false;
+        if (expr instanceof NegativeNode && ((NegativeNode) expr).getValue() instanceof ConstantNode) return false;
```

Comment:
Shouldn't this be 
`if (expr instanceof NegativeNode && ! shouldGenerateFeature((NegativeNode) expr).getValue())) return false;
`
?

Reply:
Not really. This captures the fact that the expression parsing represents a negative constant as a `NegativeNode` containing a `ConstantNode`. So that can be output safely as is. Previous parsing (old `arg`) had a regular `ConstantNode` with negative value.

A `NegativeNode` in general should probably be in an expression. For instance `-attribute(field)` (`NegativeNode` with a `ReferenceNode`).